app/controllers/application_controller.rb

```diff
@@ -19,6 +19,6 @@ class ApplicationController < ActionController::Base
 
   def deny_access
     flash[:warning] = helpers.t('errors.not_authorized')
-    redirect_to :root
+    redirect_to main_app.root_path
```

Comment:
Without this change, the redirects from unauthorized mission control requests will loop as the mission control controller tries to hit its own root path instead of the app's.

Reply:
Whoops.